enhancements/machine-config/on-cluster-layering.md

```diff
@@ -0,0 +1,1137 @@
+
+| **Annotation Name:**                             | **Description:**                                                                                                                           |
+| ------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| `machineconfiguration.openshift.io/rebuildImage` | When present, this will clear any failed image builds (and any supporting transitory objects created) and retry the build.                 |
+| `machineconfiguration.openshift.io/noRollout`    | When present, this will perform the image build, but will not roll out the built image to the nodes within the targeted MachineConfigPool. |
+
+MachineOSConfigs can also be augmented with the following annotations:
+
+| **Annotation Name:**                             | **Description:**                                                                                                                           |
+| ------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| `machineconfiguration.openshift.io/rebuildImage` | When present, this will clear any failed image builds (and any supporting transitory objects created) and retry the build.                 |
+| `machineconfiguration.openshift.io/noRollout`    | When present, this will perform the image build, but will not roll out the built image to the nodes within the targeted MachineConfigPool. |
+
+
+#### MachineOSBuild
+
+A MachineOSBuild represents a given OS image build that was performed. This
+is created by the BuildController in response to a MachineOSConfig change or
+a MachineConfigPool change. It is expected that a MachineOSConfig will have
+multiple MachineOSBuilds associated with it. However, only one actively
+running MachineOSBuild per MachineOSConfig will be permitted at any one time.
+
+It is also worth mentioning that the name for each MachineOSBuild will be
+produced as a hash of the current MachineOSConfig state, the current
+rendered MachineConfig for the MachineConfigPool, as well as the contents of
+the `machine-config-osimageurl` ConfigMap.
+
+| **Field Name / Path (JSONPath):** | **Optional or Required:** | **Intended Use:**                                                             |
+| --------------------------------- | ------------------------- | ----------------------------------------------------------------------------- |
+| `.spec.desiredConfig`             | Required                  | Refers to the rendered MachineConfig that was used to produce the build.      |
+| `.spec.machineOSConfig`           | Required                  | Refers to the name of the MachineOSConfig that was used to produce the build. |
+
+#### MachineOSImage
```

Comment:
What is the intended consumer of this API? Is this just the status of the build as a different API?